### PR TITLE
SEC-4632: M1 Installation Updates

### PR DESCRIPTION
**Pull request by eswidler**
Tested with my laptop and on a fresh M1 laptop.  Verified that the installation script allowed me to run Leapp sessions at the end.  The rollback script would allow me to re-run the setup again from scratch.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -105,7 +110,7 @@ if [ -d "$LEAPP" ]; then
 
     # If we found at least one available session, then we can presume
     # this installation was successful.
-    if (( $(echo "$AVAILABLE_LEAPP_SESSIONS" | wc -l) > 10 )); then
+    if (( $(echo "$AVAILABLE_LEAPP_SESSIONS" | wc -l) > 0 )); then
```
Comment (eswidler) [inline setup.sh:113]: This was a typo while testing.  We're looking for at least 1 session to be loaded to consider this a success.

File: setup.sh [hunk 2]
```diff
@@ -23,20 +28,20 @@ brew install python
 brew install awscli
 
 # If using an M1 machine, add a symlink for the AWS credential files to where Leapp expects them
-if [[ ! -f /usr/local/bin/aws ]]; then
-    ln -s /opt/homebrew/bin/aws /usr/local/bin/aws
+if [[ $(uname -m) == 'arm64' ]]; then
```
Comment (eswidler) [inline setup.sh:31]: This is a more appropriate check for M1 machines.

File: setup.sh [hunk 3]
```diff
@@ -8,13 +8,18 @@ if [[ -z "${LEAPP_ROLES}" ]]; then
...
 
-# Install the xcode Command Line Tools, if they are not installed
-xcode-select --install
-
 # Install Homebrew if not installed
+# This may optionally install the Xcode CLT if it is not already installed.
 which -s brew
 if [[ $? != 0 ]] ; then
-    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" </dev/null
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
Comment (eswidler) [inline setup.sh:15]: The pipe to `/dev/null` was actually preventing the prompt for the user's password to show up when installing homebrew, so I removed that.

File: setup.sh [hunk 4]
```diff
@@ -23,20 +28,20 @@ brew install python
...
-    ln -s /opt/homebrew/bin/aws /usr/local/bin/aws
+if [[ $(uname -m) == 'arm64' ]]; then
+    sudo ln -s /opt/homebrew/bin/aws /usr/local/bin/aws
 fi
 
-# For the app store version of filezilla, it expects the .aws credentials
+# If the app store version of filezilla is installed, it expects the .aws credentials
 # to be in the filezilla installation directory.  Add a symlink there.
-ln -s ~/.aws ~/Library/Containers/org.filezilla-project.filezilla.sandbox/Data/.aws
+if [ -d ~/Library/Containers/org.filezilla-project.filezilla.sandbox ]; then
```
Comment (eswidler) [inline setup.sh:37]: Added a check for the app store installation of filezilla.

File: setup.sh [hunk 5]
```diff
@@ -23,20 +28,20 @@ brew install python
...
+    ln -s ~/.aws ~/Library/Containers/org.filezilla-project.filezilla.sandbox/Data/.aws
+fi
 
 # Install session manager plugin
-curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac/sessionmanager-bundle.zip" -o "sessionmanager-bundle.zip"
-unzip -o sessionmanager-bundle.zip
-python3 sessionmanager-bundle/install -i /usr/local/sessionmanagerplugin -b /usr/local/bin/session-manager-plugin
-rm -rf sessionmanager-bundle
+brew install --cask session-manager-plugin
```
Comment (eswidler) [inline setup.sh:42]: Switched the Session manager plugin installation to use homebrew instead.

**Review by TorresJosh — APPROVED** [review]

**Pull request merged**